src/Microsoft.Extensions.ServiceDiscovery/ServiceEndpointWatcher.cs

```diff
@@ -143,13 +156,12 @@ private async Task RefreshAsyncInternal()
                 // Check if we need to poll for updates or if we can register for change notification callbacks.
                 if (endpoints.ChangeToken.ActiveChangeCallbacks)
                 {
-                    // Initiate a background refresh, if necessary.
-                    endpoints.ChangeToken.RegisterChangeCallback(static state => _ = ((ServiceEndpointWatcher)state!).RefreshAsync(force: false), this);
-                    if (_pollingTimer is { } timer)
-                    {
-                        _pollingTimer = null;
-                        timer.Dispose();
-                    }
+                    // Initiate a background refresh when the change token fires.
+                    _changeTokenRegistration = endpoints.ChangeToken.RegisterChangeCallback(static state => _ = ((ServiceEndpointWatcher)state!).RefreshAsync(force: false), this);
```

Comment:
I know we had this before but this is async void

Reply:
It's not `async`, though